pilot/test/integration/testdata/ingress.yaml.tmpl

```diff
@@ -41,6 +41,22 @@ spec:
         backend:
           serviceName: b
           servicePort: 80
+      - path: /javastyle/*
+          backend:
+            serviceName: a
+            servicePort: 80
+      - path: /.well_known/foo
+          backend:
+            serviceName: a
+            servicePort: 80
+      - path: /.well_known/foo/prefix/*
```

Comment:
This might interpolate well in the shell, but in regex world, this has strange semantics. 0 or more occurrences of `/` ..

Reply:
That's actually a feature :-), since multiple // in URLs are not good.

But the syntax is what Java Servlet spec has used for >10 years.  
I don't mind using the go ServeMux syntax ( with just "url/" ) - since it's easy to differentiate
from the exact match not ending with '/'

The real problem is that it's very easy to get it wrong and send to the wrong place - regex are 
very powerful, but in majority of cases a prefix mapping is easier to understand and harder to get 
wrong. Golang and java don't support regex at all by default - so at least for services using 
standard java  or default golang it'll be more confusing to use one style of rules in the code, 
and a different one in the istio ingress.